pkgs/development/libraries/science/math/zn_poly/default.nix

```diff
@@ -0,0 +1,47 @@
+{ stdenv
+, fetchurl
+, gmp
+, python2
+}:
+
+stdenv.mkDerivation rec {
+  version = "0.9";
+  pname = "zn_poly";
+  name = "${pname}-${version}";
+
+  src = fetchurl {
+    url = "http://web.maths.unsw.edu.au/~davidharvey/code/zn_poly/releases/zn_poly-${version}.tar.gz";
+    sha256 = "1kxl25av7i3v68k32hw5bayrfcvmahmqvs97mlh9g238gj4qb851";
+  };
+
+  buildInputs = [
+    gmp
+  ];
+
+  nativeBuildInputs = [
+    python2 # needed by ./configure to create the makefile
+  ];
+
+  # Tuning (either autotuning or with hand-written paramters) is possible
+  # but not implemented here.
+  # It seems buggy anyways (see homepage).
+  buildFlags = "all libzn_poly.so";
+
+  installPhase = ''
+    mkdir -p "$out/include/zn_poly"
+    mkdir -p "$out/lib"
+    ls
```

Comment:
This should be removed.

Also check to make sure "make install" doesn't work- it's okay to supply installPhase just usually a lot harder to maintain.

Reply:
Yes, its necessary. For some reason the regular `make install` does not install some header files and the so file. I've added a comment explaining this (and removed the stray `ls`).